src/components/global-context/cart-context.js

```diff
@@ -0,0 +1,46 @@
+import axios from "axios";
+import React, { useEffect, createContext, useContext, useReducer } from "react";
+import { useAuthContext } from "./auth-context";
+import { useGlobalVarContext } from "./global-variables";
+
+const CartContext = createContext();
+
+const cartReducer = (state, action) => {
+  switch (action.type) {
+    case "GET_CARTLIST":
+      return { ...state, cartlist: action.payload };
+
+    default:
+      break;
+  }
+};
+
+const CartContextProvider = ({ children }) => {
+  const { token } = useGlobalVarContext();
+  const [state, dispatch] = useReducer(cartReducer, { cartlist: [] });
+  const { isUserAuthenticated } = useAuthContext();
+  const getCartlist = async (encToken) => {
```

Comment:
As a best practice, API calls can be put in services as well

Reply:
Will do that , thanks for pointing out